### PR TITLE
Interop tests

### PR DESCRIPTION
**Pull request by Menduist**
Waiting on https://github.com/libp2p/test-plans/pull/138

Will test a simple ping with every go & rust version, for every transport, muxer & encryption combo

### CODE REVIEW

File: .github/workflows/interop.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          repository: status-im/libp2p-test-plans
+          path: test-plans
+          ref: nimsupport
+          submodules: true
+          fetch-depth: 0
+
+      - name: Build image
+        run: >
+          cd test-plans/multidim-interop/nim/v1.0 &&
+          make commitSha=$GITHUB_SHA image_name=nim-libp2p-head
```
Comment (MarcoPolo) [inline .github/workflows/interop.yml:27]: Cool! is this the commit sha for the current PR?

File: .github/workflows/interop.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+      - name: Create ping-version.json
+        run: >
+          echo """
+            {
+                "id": "nim-libp2p-head",
+                "containerImageID": "nim-libp2p-head",
+                "transports": [
+                    "tcp",
+                    "ws"
```
Comment (MarcoPolo) [inline .github/workflows/interop.yml:37]: Nice! Lets you have a single file. Don't forget to update this when you add new capabilities :)

File: .github/workflows/interop.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                ],
+                "secureChannels": [
+                    "noise"
+                ],
+                "muxers": [
+                    "mplex",
+                    "yamux"
+                ]
+            }
+          EOF
```
Comment (Menduist) [inline .github/workflows/interop.yml:47]:
```suggestion
          EOF
          
```

File: .github/workflows/interop.yml (hunk 1 above)
Comment (Menduist) [reply .github/workflows/interop.yml:27]: Yes :)
https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables

**Review by lchenut — APPROVED** [review]

**Pull request merged**